src/app/main.rs

```diff
@@ -420,13 +421,20 @@ where
                     .push(buffer::layer(MAX_IN_FLIGHT))
                     .push(timeout::layer(config.bind_timeout))
                     .push(limit::layer(MAX_IN_FLIGHT))
+                    .push(strip_header::layer(super::DST_OVERRIDE_HEADER))
                     .push(router::layer(|req: &http::Request<_>| {
                         let addr = super::http_request_authority_addr(req)
                             .or_else(|_| super::http_request_host_addr(req))
                             .or_else(|_| super::http_request_orig_dst_addr(req))
                             .ok();
                         debug!("outbound addr={:?}", addr);
-                        addr
+                        super::http_request_l5d_override_dst_addr(req)
```

Comment:
If the override always takes precedent, perhaps we should check it first, before the other checks?

Reply:
@seanmonstar that was the original implementation (see 2493c85): https://github.com/linkerd/linkerd2-proxy/blob/2493c85d1c091afea2e44129fa8377d7fedd5517/src/app/main.rs#L423-L431
I thought that for debugging purposes, it would be valuable to log the non-overridden addr _as well_ as the overridden one (similar to how the inbound router logs both the value of the `l5d-canonical-dst` header _and_ the final dst addr), so I encouraged @kleimkuhler to rewrite this closure so that we always determine the pre-override addr before checking for the override.

I think this can be changed back if you don't think logging both is necessary.

Reply:
The logging *could* be useful. In that case, though, it can be structured so that the extra parsing only occurs if actually logging at `debug` level. That way, if no one is looking, why pay extra.

Reply:
Okay, in that case, I propose something like:
```rust
 .push(router::layer(|req: &http::Request<_>| { 
     let addr = || {
         super::http_request_authority_addr(req)) 
            .or_else(|_| super::http_request_host_addr(req)) 
            .or_else(|_| super::http_request_orig_dst_addr(req)) 
            .ok(); 
     };
    super::http_request_l5d_override_dst_addr(req) 
        .map(|override_addr| {
            if log_enabled!(log::Level::Debug) {
                debug!("outbound addr={:?}; override_dst={:?}", addr(), override_addr); 
            }
            override_addr
        })
        .or_else(|_| {
            let addr = addr();
            debug!("outbound addr={:?}", addr);
            addr
        })
 })
```
This way, the pre-override addr is only computed if we're actually going to log it.

Unfortunately this does make the code a tad more complex...perhaps some of this could be factored out into functions or something.

Reply:
Would we still want to keep the log line for `dst-override` outside of `log_enabled!`?